psydac/api/settings.py

```diff
@@ -1,39 +1,38 @@
 # coding: utf-8
-
-# ... Determine Pyccel version: compiler names changed with version 1.3.0
-import pyccel
-pyccel_version = tuple(map(int, pyccel.__version__.split('.')))
-pyccel_legacy  = pyccel_version < (1, 3, 0)
-# ...
+import platform
 
 
 PSYDAC_DEFAULT_FOLDER = {'name':'__psydac__'}
 
 # ... defining PSYDAC backends
 PSYDAC_BACKEND_PYTHON = {'name': 'python', 'tag':'python', 'openmp':False}
 
-PSYDAC_BACKEND_GPYCCEL = {'name':     'pyccel',
-                      'compiler': 'gfortran' if pyccel_legacy else 'GNU',
-                      'flags':    '-O3 -march=native -mtune=native  -mavx -ffast-math -ffree-line-length-none',
-                      'folder': '__gpyccel__',
-                      'tag':'gpyccel',
-                      'openmp':False}
-
-PSYDAC_BACKEND_IPYCCEL = {'name':     'pyccel',
-                      'compiler': 'ifort' if pyccel_legacy else 'intel',
-                      'flags':    '-O3',
-                      'folder': '__ipyccel__',
-                      'tag':'ipyccel',
-                      'openmp':False}
-
-PSYDAC_BACKEND_PGPYCCEL = {'name':     'pyccel',
-                      'compiler': 'pgfortran' if pyccel_legacy else 'PGI',
-                      'flags':    '-O3',
-                      'folder': '__pgpyccel__',
-                       'tag':'pgpyccel',
-                       'openmp':False}
+PSYDAC_BACKEND_GPYCCEL  = {'name': 'pyccel',
+                       'compiler': 'GNU',
+                       'flags'   : '-O3 -march=native -mtune=native -ffast-math -ffree-line-length-none',
+                       'folder'  : '__gpyccel__',
+                       'tag'     : 'gpyccel',
+                       'openmp'  : False}
+
+PSYDAC_BACKEND_IPYCCEL  = {'name': 'pyccel',
+                       'compiler': 'intel',
+                       'flags'   : '-O3',
+                       'folder'  : '__ipyccel__',
+                       'tag'     :'ipyccel',
+                       'openmp'  : False}
+
+PSYDAC_BACKEND_PGPYCCEL = {'name': 'pyccel',
+                       'compiler': 'PGI',
```

Comment:
PGI is a legacy compiler whose last community release was in 2019. As PGI compilers and tools have evolved into the NVIDIA HPC SDK, it might be a good idea to add a nvidia option.

Reply:
Good idea! Should we remove the option from Pyccel too? See [here](https://github.com/pyccel/pyccel/blob/b03c874705c0b45a3037fd7536a6e2e283094fd7/pyccel/compilers/default_compilers.py#L65-L80).

Reply:
I think we could consider it. As we move to version 2.0 it is a sensible time to make breaking changes like this. That said it is not difficult to support both as the only difference is the name. The reason I mentioned it here was that you are only supporting the legacy version not the supported version

Reply:
I've added the Nvidia compiler in e2df18f.